iceoryx2-cli/iox2-config/src/commands.rs

```diff
@@ -0,0 +1,127 @@
+use std::path::Path;
+
+/// Prints the whole system configuration with all limits, features and details to the console.
+pub fn print_system_configuration() {
+    const HEADER_COLOR: &str = "\x1b[4;92m";
+    const VALUE_COLOR: &str = "\x1b[0;94m";
+    const DISABLED_VALUE_COLOR: &str = "\x1b[0;90m";
+    const ENTRY_COLOR: &str = "\x1b[0;37m";
+    const DISABLED_ENTRY_COLOR: &str = "\x1b[0;90m";
+    const COLOR_RESET: &str = "\x1b[0m";
+
+    println!("{}posix system configuration{}", HEADER_COLOR, COLOR_RESET);
+    println!();
+    println!(" {}system info{}", HEADER_COLOR, COLOR_RESET);
+    for i in all::<SystemInfo>().collect::<Vec<_>>() {
+        println!(
+            "  {ENTRY_COLOR}{:<50}{COLOR_RESET} {VALUE_COLOR}{}{COLOR_RESET}",
+            format!("{:?}", i),
+            i.value(),
+        );
+    }
+
+    println!();
+    println!(" {}limits{}", HEADER_COLOR, COLOR_RESET);
+    for i in all::<Limit>().collect::<Vec<_>>() {
+        let limit = i.value();
+        let limit = if limit == 0 {
+            "[ unlimited ]".to_string()
+        } else {
+            limit.to_string()
+        };
+        println!(
+            "  {ENTRY_COLOR}{:<50}{COLOR_RESET} {VALUE_COLOR}{}{COLOR_RESET}",
+            format!("{:?}", i),
+            limit,
+        );
+    }
+
+    println!();
+    println!(" {}options{}", HEADER_COLOR, COLOR_RESET);
+    for i in all::<SysOption>().collect::<Vec<_>>() {
+        if i.is_available() {
+            println!(
+                "  {ENTRY_COLOR}{:<50}{COLOR_RESET} {VALUE_COLOR}{}{COLOR_RESET}",
+                format!("{:?}", i),
+                i.is_available(),
+            );
+        } else {
+            println!(
+                "  {DISABLED_ENTRY_COLOR}{:<50}{COLOR_RESET} {DISABLED_VALUE_COLOR}{}{COLOR_RESET}",
+                format!("{:?}", i),
+                i.is_available(),
+            );
+        }
+    }
+
+    println!();
+    println!(" {}features{}", HEADER_COLOR, COLOR_RESET);
+    for i in all::<Feature>().collect::<Vec<_>>() {
+        if i.is_available() {
+            println!(
+                "  {ENTRY_COLOR}{:<50}{COLOR_RESET} {VALUE_COLOR}{}{COLOR_RESET}",
+                format!("{:?}", i),
+                i.is_available(),
+            );
+        } else {
+            println!(
+                "  {DISABLED_ENTRY_COLOR}{:<50}{COLOR_RESET} {DISABLED_VALUE_COLOR}{}{COLOR_RESET}",
+                format!("{:?}", i),
+                i.is_available(),
+            );
+        }
+    }
+
+    println!();
+    println!(" {}process resource limits{}", HEADER_COLOR, COLOR_RESET);
+    for i in all::<ProcessResourceLimit>().collect::<Vec<_>>() {
+        println!(
+            "  {ENTRY_COLOR}{:<43}{COLOR_RESET} soft:  {VALUE_COLOR}{:<24}{COLOR_RESET} hard:  {VALUE_COLOR}{}{COLOR_RESET}",
+            format!("{:?}", i),
+            i.soft_limit(),
+            i.hard_limit()
+        );
+    }
+}
+
+pub fn show() -> Result<()> {
+    print_system_configuration();
```

Comment:
Does this print the default in-built config if there is no config file being loaded from file?

Reply:
print_system_configuration() displays the whole system configuration with all limits, features and details to the console. It does not print the built-in config but the system configuration. print_system_configuration() function is independent of the config file.

Reply:
@elfenpiff Should the defaults be added to the output of `iox2 config show`, under a different sub-heading ? I would think yes, but what did you have in mind?

Reply:
@elfenpiff Will need your feedback on this. What was your intention for this command?

Reply:
@elfenpiff I implemented this to provide me with some debug output of the configuration of the POSIX system, meaning the POSIX configuration that states like how many semaphores can be created or what the maximum value of a semaphore could be.

Windows, Mac Os, FreeBSD, and Linux have different values here, and in the early days of iceoryx2, it was very helpful when analyzing bugs. Now, everything is abstracted away with the `iceoryx2_pal` layer but this still makes sense for mission-critical environments where we define the minimum requirements we have for the system (somewhere) and then this would print out the systems requirement and would also highlight the things which would violate our requirements.

Reply:
@elfenpiff Apologies, what is your advice here ?

Reply:
@orecham My apologies!

> Should the defaults be added to the output of iox2 config show, under a different sub-heading ? 

Yes, this would make sense.

Reply:
@elfenpiff My apologies, did you say I had to set up a new subtitle? Like “iox2 config show default” which shows the default configuration? Am I right ?

Reply:
@brosier01 Yeah, if you could additionally print the contents of the config file under another subtitle, this would be perfect.

Let us know if you hit problems or it turns out more difficult than it seems.